tests/unit/lms/views/api/lti_test.py

```diff
@@ -122,13 +128,14 @@ def pyramid_request(self, pyramid_request):
 
 
 class TestRecordResult:
-    def test_it_records_result(self, pyramid_request, lti_outcomes_client):
+    def test_it_records_result(self, ai_getter, pyramid_request, lti_outcomes_client):
 
         LTIOutcomesViews(pyramid_request).record_result()
 
+        ai_getter.shared_secret.assert_called_once_with("TEST_OAUTH_CONSUMER_KEY")
```

Comment:
This assertion was previously hidden in the body of the `ai_getter` fixture (see below) so you couldn't see, just from looking the `TestRecordResult` tests themselves, whether this was actually being tested. You would have to read the code body of the `ai_getter` fixture itself. And if we had a habit of putting our assertions inside fixtures then you would have to read the code of every one of this test's fixtures, and every fixture of every test, to know what is being tested. And the more complex fixtures would tangle all the different tests together. Just put your assertions in your tests

Reply:
Principal of least surprise is a good thing